Common/hipblaslt_utils.hpp

```diff
@@ -0,0 +1,1051 @@
+
+#ifdef _WIN32
+    #include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
+
+/// \brief Converts a \p hipblasStatus_t variable to its correspondent string.
+inline const char* hipblas_status_to_string(hipblasStatus_t status)
+{
+    switch(status)
+    {
+        case HIPBLAS_STATUS_SUCCESS: return "HIPBLAS_STATUS_SUCCESS";
+        case HIPBLAS_STATUS_NOT_INITIALIZED: return "HIPBLAS_STATUS_NOT_INITIALIZED";
+        case HIPBLAS_STATUS_ALLOC_FAILED: return "HIPBLAS_STATUS_ALLOC_FAILED";
+        case HIPBLAS_STATUS_INVALID_VALUE: return "HIPBLAS_STATUS_INVALID_VALUE";
+        case HIPBLAS_STATUS_MAPPING_ERROR: return "HIPBLAS_STATUS_MAPPING_ERROR";
+        case HIPBLAS_STATUS_EXECUTION_FAILED: return "HIPBLAS_STATUS_EXECUTION_FAILED";
+        case HIPBLAS_STATUS_INTERNAL_ERROR: return "HIPBLAS_STATUS_INTERNAL_ERROR";
+        case HIPBLAS_STATUS_NOT_SUPPORTED: return "HIPBLAS_STATUS_NOT_SUPPORTED";
+        case HIPBLAS_STATUS_ARCH_MISMATCH: return "HIPBLAS_STATUS_ARCH_MISMATCH";
+        case HIPBLAS_STATUS_HANDLE_IS_NULLPTR: return "HIPBLAS_STATUS_HANDLE_IS_NULLPTR";
+        case HIPBLAS_STATUS_INVALID_ENUM: return "HIPBLAS_STATUS_INVALID_ENUM";
+        case HIPBLAS_STATUS_UNKNOWN: return "HIPBLAS_STATUS_UNKNOWN";
+        // We use default because we are not in control of these enumeration values.
+        // Ideally this function is something hipBLAS would provide
+        default: return "<unknown hipblasStatus_t value>";
+    }
+}
+
+/// \brief Checks if the provided status code is \p HIPBLAS_STATUS_SUCCESS and if not,
+/// prints an error message to the standard error output and terminates the program
+/// with an error code.
+#define HIPBLASLT_CHECK(condition)                                                             \
+    {                                                                                          \
+        const hipblasStatus_t status = (condition);                                            \
+        if(status != HIPBLAS_STATUS_SUCCESS)                                                   \
+        {                                                                                      \
+            std::cerr << "hipBLASLt error encountered: \"" << hipblas_status_to_string(status) \
+                      << "\" at " << __FILE__ << ':' << __LINE__ << std::endl;                 \
+            std::exit(error_exit_code);                                                        \
+        }                                                                                      \
+    }
+
+// ============================================================================
+// Data Type Interface
+// ============================================================================
+
+union compute_type_interface
+{
+    float         f32;
+    double        f64;
+    hipblasLtHalf f16;
+    int32_t       i32;
+};
+
+template<typename T>
+constexpr auto hipblaslt_type_to_datatype()
+{
+    if(std::is_same<T, hipblasLtHalf>{})
```

Comment:
```suggestion
    if(std::is_same_v<T, hipblasLtHalf>)
```

And the same for the following values.